### PR TITLE
Nighthawk: ASAN/TSAN, Update Envoy, sync up misc changes 

### PR DESCRIPTION
**Pull request by oschaaf**
- This adds ASAN/TSAN to CI testing
- Syncs a few small improvements from the NH branch
- Updates Envoy to the latest
- Envoy stopped incrementing upstream_rq_total for HTTP/1 conn pool
  when request is circuit broken. Fix test expectations.

### CODE REVIEW

File: ci/do_ci.sh [hunk 1]
```diff
@@ -23,37 +23,101 @@ function do_coverage() {
...
+
+function setup_clang_toolchain() {
+    export PATH=/usr/lib/llvm-7/bin:$PATH
+    export CC=clang
+    export CXX=clang++
+    export ASAN_SYMBOLIZER_PATH=/usr/lib/llvm-7/bin/llvm-symbolizer
+    echo "$CC/$CXX toolchain configured"
+}
+
+# This overrides both ld and ld.gold with llvm's ld.lld.
```
Comment (htuch) [inline ci/do_ci.sh]: This is an insane (in both the good and bad way ;) ) hack; kudos for figuring out a way to get this to work. OTOH, I think this should be tagged with a big TODO and a tracking issue, at some point we want to unwind this and figure out why this isn't working as expected.

File: nighthawk/test/utility_test.cc [hunk 2]
```diff
@@ -128,6 +129,20 @@ TEST_P(UtilityAddressResolutionTest, AddressResolutionBadAddresses) {
   EXPECT_THROW(testResolution("a..b", address_family), UriException);
 }
 
+TEST_P(UtilityAddressResolutionTest, resolveTwiceReturnsCached) {
```
Comment (htuch) [inline nighthawk/test/utility_test.cc]: Same nit as in other PR about naming tests, starting with capital letter.

File: ci/do_ci.sh (hunk 1 above)
Comment (oschaaf) [reply ci/do_ci.sh]: Done

File: nighthawk/test/utility_test.cc (hunk 2 above)
Comment (oschaaf) [reply nighthawk/test/utility_test.cc]: Done

**Pull request merged**